MergerLogic/DataTypes/S3.cs

```diff
@@ -97,18 +87,7 @@ public override List<Tile> GetNextBatch(out string currentBatchIdentifier, out s
                     }
 
                     string path = $"{this.Path}/{this._zoomEnumerator.Current}/";
-                    var listRequests = new ListObjectsV2Request
-                    {
-                        BucketName = this._bucket,
-                        Prefix = path,
-                        StartAfter = path,
-                        MaxKeys = missingTiles,
-                        ContinuationToken = this._continuationToken
-                    };
-
-                    var listObjectsTask = this._client.ListObjectsV2Async(listRequests);
-                    var response = listObjectsTask.Result;
-
+                    ListObjectsV2Response response = this.Utils.ListObject(ref this._continuationToken, path, path, missingTiles);
```

Comment:
I understand the thought, but because of the generic use this is a mistake.
You could move this function to this class instead.